Objects/genobject.c

```diff
@@ -32,6 +32,8 @@ gen_traverse(PyGenObject *gen, visitproc visit, void *arg)
     Py_VISIT(gen->gi_code);
     Py_VISIT(gen->gi_name);
     Py_VISIT(gen->gi_qualname);
+    if (((PyCodeObject *)gen->gi_code)->co_flags & CO_COROUTINE)
+        Py_VISIT(((PyCoroObject *)gen)->cr_origin);
```

Comment:
If you only store ints and strings in tuples of that list (without cycles), you can skip the `Py_VISIT` call.